doc/shared/suncontrol/SUNControl_Description.rst

```diff
@@ -0,0 +1,480 @@
+..
+   ----------------------------------------------------------------
+   SUNDIALS Copyright Start
+   Copyright (c) 2002-2023, Lawrence Livermore National Security
+   and Southern Methodist University.
+   All rights reserved.
+
+   See the top-level LICENSE and NOTICE files for details.
+
+   SPDX-License-Identifier: BSD-3-Clause
+   SUNDIALS Copyright End
+   ----------------------------------------------------------------
+
+.. _SUNControl.Description:
+
+The SUNControl API
```

Comment:
How about `SUNAdaptController` and then the heuristics class could be `SUNAdaptHeuristics`.

Reply:
Given that "controller" has a specific connotation in this field regarding control-theoretic error-based adaptivity, I feel like extending the name to `SUNAdaptController` only increases typing burden.  Given that some of these function names are already incredibly long, I am reticent to make it longer unnecessarily.

That said, there are many forms of heuristics throughout SUNDIALS, so the base name `SUNHeuristics` may not be sufficiently specific (although it's already long enough).  Since it focuses on step size heuristics, then what do you think about `SUNTimestepHeuristics`?

Reply:
I like either `SUNTImestepHeuristics` or `SUNAdaptHeuristics`.  I dislike `SUNControl` but would be OK with `SUNController`. However, I most prefer `SUNAdaptController` or `SUNTimestepController` but I tend to prefer descriptive names over saving a few keystrokes.  Also, not every user of SUNDIALS is in tune with the time integration field and they might not associate "controller" with time-step adaptivity automatically like we do.

Reply:
I'll go with `SUNAdaptController` and `SUNTimestepHeuristics`